File: rick.pas
program Rickroll;

uses
  SysUtils, Process;

procedure OpenRickroll;
var
  url: string;
begin
  url := 'https://www.youtube.com/watch?v=dQw4w9WgXcQ';
  if Pos('Win', {$IFDEF MSWINDOWS} 'Windows' {$ELSE} '' {$ENDIF}) > 0 then
    SysUtils.ShellExecute(0, 'open', PChar(url), nil, nil, SW_SHOWNORMAL)
  else if Pos('Mac', {$IFDEF MACOS} 'Mac' {$ELSE} '' {$ENDIF}) > 0 then
    SysUtils.ShellExecute(0, 'open', PChar(url), nil, nil, SW_SHOWNORMAL)
  else
    SysUtils.ShellExecute(0, 'open', PChar(url), nil, nil, SW_SHOWNORMAL);  // Assume Linux
end;

var
  output: string;
  return_code: Integer;
begin
  output := RunCommand('curl ascii.live/rick', return_code);

  if return_code = 0 then
    Writeln(output)  // Print command output
  else
  begin
    Writeln('Command failed.');
    OpenRickroll();
  end;
end.

File: rick.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.awt.Desktop
import java.net.URI

fun main() {
    val command = arrayOf("curl", "ascii.live/rick")

    try {
        val process = ProcessBuilder(*command).start()
        val reader = BufferedReader(InputStreamReader(process.inputStream))
        val output = reader.readText()

        if (process.waitFor() == 0) {
            println(output) // Print command output
        } else {
            println("Command failed.")
            openRickroll()
        }
    } catch (e: Exception) {
        println("An error occurred: ${e.message}")
        openRickroll()
    }
}

fun openRickroll() {
    val url = "https://www.youtube.com/watch?v=dQw4w9WgXcQ"
    if (Desktop.isDesktopSupported()) {
        Desktop.getDesktop().browse(URI(url))
    } else {
        println("Desktop is not supported. Open the link manually: $url")
    }
}